fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/service/GlobalConfigurationPropertyUpdateService.java

```diff
@@ -0,0 +1,100 @@
+import org.apache.fineract.portfolio.savings.data.SavingsProductData;
+import org.apache.fineract.portfolio.savings.service.SavingsProductReadPlatformService;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class GlobalConfigurationPropertyUpdateService {
+
+    private final LoanProductReadPlatformService loanProductReadPlatformService;
+    private final SavingsProductReadPlatformService savingsProductReadPlatformService;
+
+    public Map<String, Object> update(final GlobalConfigurationProperty property, final JsonCommand command) {
+        final Map<String, Object> actualChanges = new LinkedHashMap<>(7);
+
+        if (property.isTrapDoor() && isAnyProductAlreadyCreated()) {
+            throw new GlobalConfigurationPropertyCannotBeModfied(property.getId());
+        }
+
+        final String enabledParamName = "enabled";
+        if (command.isChangeInBooleanParameterNamed(enabledParamName, property.isEnabled())) {
+            final boolean newValue = command.booleanPrimitiveValueOfParameterNamed(enabledParamName);
+            actualChanges.put(enabledParamName, newValue);
+            property.setEnabled(newValue);
+        }
+
+        final String valueParamName = "value";
+        final Long previousValue = property.getValue();
+        if (command.isChangeInLongParameterNamed(valueParamName, property.getValue())) {
+            final Long newValue = command.longValueOfParameterNamed(valueParamName);
+            actualChanges.put(valueParamName, newValue);
+            property.setValue(newValue);
+        }
+
+        final String dateValueParamName = "dateValue";
+        if (command.isChangeInDateParameterNamed(dateValueParamName, property.getDateValue())) {
+            final LocalDate newDateValue = command.localDateValueOfParameterNamed(dateValueParamName);
+            actualChanges.put(dateValueParamName, newDateValue);
+            property.setDateValue(newDateValue);
+        }
+
+        final String stringValueParamName = "stringValue";
+        if (command.isChangeInStringParameterNamed(stringValueParamName, property.getStringValue())) {
+            final String newStringValue = command.stringValueOfParameterNamed(stringValueParamName);
+            actualChanges.put(stringValueParamName, newStringValue);
+            property.setStringValue(newStringValue);
+        }
+
+        final String passwordPropertyName = GlobalConfigurationConstants.FORCE_PASSWORD_RESET_DAYS;
+        if (property.getName().equalsIgnoreCase(passwordPropertyName)) {
+            if ((property.isEnabled() && command.hasParameter(valueParamName) && (property.getValue() == 0))
+                    || (property.isEnabled() && !command.hasParameter(valueParamName) && (previousValue == 0))) {
+                throw new ForcePasswordResetException();
+            }
+        }
+
+        return actualChanges;
+    }
+
+    private boolean isAnyProductAlreadyCreated() {
```

Comment:
I would rather like an option where:
we are checking only the existence of any "loan (m_loan)", "savings (m_savings_account)", or "shares (m_share_account)".
`SELECT COUNT(1) ...` is more than enough, we dont need to fetch any data!

Reply:
sure, so I should just take a jdbcTemplate and check the row counts in those tables right?

Reply:
Alternatively you can use the `CriteriaQueryFactory` to build a query which do the same...

Reply:
I've pushed the fix, in the end I chose JdbcTemplate as it seemed simpler for this use-case